sql: TestDockerCLI/test_txn_prompt.tcl is broken on master

This is reproducible locally - `make acceptance TESTS=TestDockerCLI/test_txn_prompt`

The failure log is here: https://gist.github.com/jordanlewis/edd8b3fe10e6172831bdf48cfa038780

cc @knz @tamird

Also, the failure message is entirely inscrutable (to me). Here's the test run:
```
I170130 11:49:54.161242 1 rand.go:76  Random seed: -3517688313855384090
CockroachDB node starting at 2017-01-30 16:50:05.677073382 +0000 UTC
build:      CCL e105ffe @ 2017/01/30 16:24:03 (go1.7.4)
admin:      http://localhost:8080
sql:        postgresql://root@localhost:26257?sslmode=disable
logs:       cockroach-data/logs
store[0]:   path=cockroach-data
status:     initialized new cluster
clusterID:  3de4fbd2-10d2-4aae-8754-8ae49b240d3e
nodeID:     1
spawn /bin/bash
PS1='\h:''/# '
root@d6c44da7e880:/# PS1='\h:''/# '
d6c44da7e880:/# /cockroach/cockroach sql
# Welcome to the cockroach SQL interface.
# All statements must be terminated by a semicolon.
# To exit: CTRL + D.
root@:26257/> begin;
Now adding input for a multi-line SQL transaction client-side.
Press Enter two times to send the SQL text collected so far to the server, or Ctrl+C to cancel.
           ->
BEGIN
root@:26257/  OPEN> select a;
pq: column name "a" not found
root@:26257 ERROR> rollback;
ROLLBACK
root@:26257/> begin;SAVEPOINT cockroach_restart;
Now adding input for a multi-line SQL transaction client-side.
Press Enter two times to send the SQL text collected so far to the server, or Ctrl+C to cancel.
           ->
OK
root@:26257/  OPEN> select 1;
+---+
| 1 |
+---+
| 1 |
+---+
(1 row)
root@:26257/  OPEN> RELEASE SAVEPOINT cockroach_restart;
COMMIT
root@:26257  DONE> commit;
COMMIT
root@:26257/> begin;SAVEPOINT cockroach_restart;
Now adding input for a multi-line SQL transaction client-side.
Press Enter two times to send the SQL text collected so far to the server, or Ctrl+C to cancel.
           ->
OK
root@:26257/  OPEN> SELECT CRDB_INTERNAL.FORCE_RETRY('1s':::INTERVAL);
pq: restart transaction: forced by crdb_internal.force_retry()
root@:26257 RETRY> ROLLBACK TO SAVEPOINT cockroach_restart;
connection lost; opening new connection and resetting session parameters...
dial tcp [::1]:26257: getsockopt: connection refused
connection lost; opening new connection and resetting session parameters...
error retrieving the transaction status: dial tcp [::1]:26257: getsockopt: connection refusedconnection lost; opening new connection and resetting session parameters...
root@:26257 ?> --- FAIL: TestDockerCLI (43.84s)
    --- FAIL: TestDockerCLI/test_txn_prompt.tcl (38.42s)
    	cli_test.go:73: non-zero exit code: 1
	test_log_scope.go:85: test log files left over in: /var/folders/lf/1_qqs3815tzgkhtrbrjs913m0000gn/T/logTestDockerCLI956366206
FAIL
FAIL	github.com/cockroachdb/cockroach/pkg/acceptance	43.890s
```

--- a/pkg/cli/sql.go
+++ b/pkg/cli/sql.go
@@ -365,31 +365,23 @@ func (c *cliState) doRefreshPrompts(nextState cliStateEnum) cliStateEnum {
 		return nextState
 	}
 
-	return c.refreshTransactionStatus(nextState)
-}
-
-// refreshTransactionStatus retrieves and sets the current transaction status.
-func (c *cliState) refreshTransactionStatus(nextState cliStateEnum) cliStateEnum {
 	// Query the server for the current transaction status.
 	query := makeQuery(`SHOW TRANSACTION STATUS`)
 	rows, err := query(c.conn)
 	if err != nil {
-		fmt.Fprintf(osStderr, "error retrieving the transaction status: %v", err)
-		return c.refreshDatabaseName(" ?", nextState)
+		return c.refreshPrompts(" ?", nextState)
 	}
-
 	if len(rows.Columns()) == 0 {
 		fmt.Fprintf(osStderr, "invalid transaction status")
-		return c.refreshDatabaseName(" ?", nextState)
+		return c.refreshPrompts(" ?", nextState)
 	}
-
 	val := make([]driver.Value, len(rows.Columns()))
 	err = rows.Next(val)
 	if err != nil {
 		fmt.Fprintf(osStderr, "invalid transaction status")
-		return c.refreshDatabaseName(" ?", nextState)
+		return c.refreshPrompts(" ?", nextState)
 	}
-	txnString := formatVal(val[0], false /* showPrintableUnicode */, false /* shownewLinesAndTabs */)
+	txnString := formatVal(val[0], false, false)
 
 	// Change the prompt based on the response from the server.
 	promptSuffix := " ?"
@@ -405,42 +397,13 @@ func (c *cliState) refreshTransactionStatus(nextState cliStateEnum) cliStateEnum
 	case sql.Open.String():
 		promptSuffix = "  OPEN"
 	}
-
-	return c.refreshDatabaseName(promptSuffix, nextState)
-}
-
-//This function is performing the 'refresh' action to update the cli prompt with the current
-//database in the user's context.
-func (c *cliState) refreshDatabaseName(txnStatus string, nextState cliStateEnum) cliStateEnum {
-	var dbVals [1]driver.Value
-
-	query := makeQuery(`SHOW DATABASE`)
-	rows, err := query(c.conn)
-
-	if err != nil {
-		fmt.Fprintf(osStderr, "error retrieving the database name: %v", err)
-		return c.refreshPrompts(txnStatus, nextState)
-	}
-
-	if len(rows.Columns()) == 0 {
-		fmt.Fprintf(osStderr, "cannot get the database name")
-		return c.refreshPrompts(txnStatus, nextState)
-	}
-
-	err = rows.Next(dbVals[:])
-	if err != nil {
-		return c.refreshPrompts(txnStatus, nextState)
-	}
-
-	dbName := formatVal(dbVals[0].(string), false /* showPrintableUnicode */, false /* shownewLinesAndTabs */)
-	return c.refreshPrompts("/"+dbName+txnStatus, nextState)
+	return c.refreshPrompts(promptSuffix, nextState)
 }
 
 func (c *cliState) refreshPrompts(promptSuffix string, nextState cliStateEnum) cliStateEnum {
 	c.fullPrompt = c.promptPrefix + promptSuffix
 	c.continuePrompt = strings.Repeat(" ", len(c.fullPrompt)-1) + "-> "
 	c.fullPrompt += "> "
-
 	return nextState
 }
 

